File: algo/leetcode/palindrome_number__LeetCode9/PalindromeNumberSolution.js
const isPalindrome = function(x) {
    if (x < 0){
        return false
    }
    let xCopy = x;
    let revX = 0;
    while(xCopy > 0){
        console.log(revX)

        revX *= 10
        revX += xCopy % 10
        xCopy = Math.floor(xCopy / 10)
    }
    return revX == x;
};

File: algo/leetcode/palindrome_number__LeetCode9/PalindromeNumberSolution.java
package leetcode.palindrome_number__LeetCode9;

public class PalindromeNumberSolution {
    public boolean isPalindrome(int x) {
        if (x < 0) {
            return false;
        }
        var xC = x;
        var revX = 0;
        while(xC > 0){
            revX *= 10;
            revX += xC % 10;
            xC /= 10;
        }
        return revX == x;
    }
}